pkg/image/validation/kubernetes.go

```diff
@@ -44,21 +46,49 @@ func isKubernetesDefined(k8s *image.Kubernetes) bool {
 	return k8s.Version != ""
 }
 
-func validateNodes(k8s *image.Kubernetes) []FailedValidation {
+func validateNetwork(k8s *image.Kubernetes) []FailedValidation {
 	var failures []FailedValidation
+	var vipAddress = k8s.Network.APIVIP
+
+	if vipAddress == "" {
+		numNodes := len(k8s.Nodes)
+		// Single node cluster, apiVIP is optional, check only multi-node
+		if numNodes > 1 {
+			failures = append(failures, FailedValidation{
+				UserMessage: "The 'apiVIP' field is required in the 'network' section when defining entries under 'nodes'.",
+			})
+		}
 
-	numNodes := len(k8s.Nodes)
-	if numNodes <= 1 {
-		// Single node cluster, node configurations are not required
 		return failures
 	}
 
-	if k8s.Network.APIVIP == "" {
+	ip, err := netip.ParseAddr(vipAddress)
+	if err != nil {
+		msg := fmt.Sprintf("Invalid cluster API address ('%s')", strings.ToLower(vipAddress))
 		failures = append(failures, FailedValidation{
-			UserMessage: "The 'apiVIP' field is required in the 'network' section when defining entries under 'nodes'.",
+			UserMessage: msg,
 		})
 	}
 
+	if ip.IsUnspecified() || ip.IsMulticast() {
```

Comment:
I'm wondering if these two should be separate and include the actual issue in the validation message e.g. `Cluster API address cannot be a multicast address`. What do you think?

Reply:
The rationale is to exclude anything that has special meaning and allow all the rest of the space, not just public/private address spaces. It is not that useful to distinguish them, and after adding a couple more groups and checking the source code of IsGlobalUnicast() I have realized it does exactly that. Initially I thought it only included IANA defined GlobalUnicast addresses for IPv6. Bottom line, we can just say that it's not a unicast address, it's more that understandable to people dealing with networking.